R/configure.R

```
@@ -0,0 +1,89 @@
+##'
+##' How big is an object? We serialise the object
+##'   (`redux::object_to_bin` just wraps [`serialize`]) which creates
+##'   a vector of bytes and that is saved into the database. To get an
+##'   idea of how large things are you can do:
+##'   `length(redux::object_to_bin(your_object))`.  At the time this
+##'   documentation was written, `mtcars` was `3807` bytes, and a
+##'   million random numbers was `8,000,031` bytes. It's unlikely that
+##'   a `store_max_size` of less than 1MB will be sensible.
+##'
+##' @title Configure rrq
+##'
+##' @param queue_id The queue id; the same as you would pass to
+##'   [rrq::rrq_controller]
+##'
+##' @param con A redis connection
+##'
+##' @param ... Additional arguments - this must be empty. This
+##'   argument exists so that all additional arguments must be passed
+##'   by name.
+##'
+##' @param store_max_size The maximum object size, in bytes, before
+##'   being moved to the offload store. If given, then larger data
+##'   will be saved in `offload_path` (using
+##'   [`rrq::object_store_offload_disk`])
+##'
+##' @param offload_path The path to create an offload store at (passed
+##'   to [`rrq::object_store_offload_disk`]). The directory will be
+##'   created if it does not exist. If not given (or `NULL`) but
+##'   `store_max_size` is finite, then trying to save large objects
+##'   will throw an error.
+##' 
+##' @return Invisibly, a list with processed configuration information
+##' @export
+rrq_configure <- function(queue_id, con = redux::hiredis(), ...,
+                          store_max_size = Inf, offload_path = NULL) {
+  if (length(list(...)) > 0) {
+    stop("Unconsumed dot arguments")
+  }
+  keys <- rrq_keys_common(queue_id)
+  config <- con$GET(keys$configuration)
+  if (!is.null(config)) {
+    ## We could check that it's the same?
+    stop(sprintf(
+      "Can't set configuration for queue '%s' as it already exists",
+      queue_id))
+  }
+  ## TODO: we should probably validate these.
```

On the fence about how much we should do this right now or later

---

Makes sense to validate now and fail early. Is there an advantage I am missing to doing it later?

---

just being lazy really :)